doc/source/rayserve/tutorials/batch-tutorial.rst

```diff
@@ -0,0 +1,119 @@
+.. _serve-batch-tutorial:
+
+Batching Tutorial
+=================
+
+In this guide, we will deploy a simple vectorized adder that takes
+a batch of queries and add them at once. In particular, we show:
+
+- How to implement and deploy RayServe model that accepts batches.
+- How to configure the batch size.
+- How to query the model in Python.
+
+This tutorial should help the following use cases:
+
+- You want to perform offline batch inference on a cluster of machines.
+- You want to serve online queries and your model can take advantage of batching.
+  For example, linear regressions and neural networks use CPU and GPU's
+  vectorized instructions to perform computation in parallel. Performing
+  inference with batching can increase the *throughput* of the model as well as
+  *utilization* of the hardware.
+
+
+Let's import RayServe and some other helpers.
+
+.. literalinclude:: ../../../../python/ray/serve/examples/doc/tutorial_batch.py
+    :start-after: __doc_import_begin__
+    :end-before: __doc_import_end__
+
+You can use ``@serve.accept_batch`` decorator to annotate a function or a class.
+This annotation is needed because batched backends have different APIs compared
+to single request backend. In a batched backend, the inputs are a list of values.
+
+For single query backend, the input types are single flask request or Python
+argument:
+
+.. code-block:: python
+
+    def single_request(
+        flask_request: Flask.Request,
+        *,
+        python_arg: int = 0
+    ):
+        pass
+
+For batched backend, the inputs types are converted to list of their original
+types:
+
+.. code-block:: python
+
+    @serve.accept_batch
+    def blacked_request(
+        flask_request: List[Flask.Request],
```

Comment:
Meta comment. What if you had just one `*_request` method and it always takes a list, even for single requests, where the list will have one item. A bit wasteful for single requests, but it simplifies everything else.